distr-otel: add description of the basic auth extension (#13)
max-cx/openshift-docs: modules/distr-tracing-otel-config-collector.adoc

          exporters: [logging]
----
<1> Verbosity of the logging export: `detailed` or `normal` or `basic`. When set to `detailed`, pipeline data is verbosely logged. Defaults to `normal`.

[id="extensions_{context}"]
=== Extensions

[id="basicauth-extension_{context}"]

> ==== BASICAUTH extension

Reviewer: We do have `OTLP exporter` and `OTLP HTTP exporter` because OTLP and HTTP are abbreviations, but  I think we don't need to capitalize the whole `BASICAUTH`:
```suggestion
==== BasicAuth extension
```

> The BasicAuth extension can be used as an Authenticator on receivers and exporters.

Reviewer: https://opentelemetry.io/docs/collector/custom-auth/ uses lowercase (`authenticator`).
```suggestion
The BasicAuth extension can be used as an authenticator on receivers and exporters.
```

> The BasicAuth extension can be used as an authenticator on receivers and exporters.

Reviewer: ```suggestion
You can use the BasicAuth extension as an authenticator on receivers and exporters.
```

> The configuration should specify only one instance of basicauth extension for either client or server authentication.

Reviewer: ```suggestion
The configuration must specify only one instance of the BasicAuth extension: for client authentication or server authentication.
```

> The configuration must specify only one instance of the BasicAuth extension: for client authentication or server authentication.

Reviewer: ```suggestion
Only one instance of the BasicAuth extension is supported in the OpenTelemetry Collector custom resource. Client authentication and server authentication are configured in separate sections in the OpenTelemetry Collector custom resource.
```

Author: > Only one instance of the BasicAuth extension is supported in the OpenTelemetry Collector custom resource

What do you mean by one instance?

Reviewer: I mean this: https://github.com/max-cx/openshift-docs/pull/13/commits/b7f9b8bdb88bb77ad121b7427a037c3daf1d934f#r1334068446.

Reviewer: I mean this: https://github.com/max-cx/openshift-docs/pull/13/commits/b7f9b8bdb88bb77ad121b7427a037c3daf1d934f#r1334068446

Reviewer: I mean this: https://github.com/max-cx/openshift-docs/pull/13#discussion_r1334068446

Reviewer: Looking at your original sentence:
`The configuration should specify only one instance of basicauth extension for either client or server authentication.`
Do we need this part at all: `only one instance of basicauth extension`?

Reviewer: How about this?
```suggestion
Client authentication and server authentication for the BasicAuth extension are configured in separate sections in the OpenTelemetry Collector custom resource.
```

* Support level: link:https://access.redhat.com/support/offerings/techpreview[Technology Preview]
* Supported signals: traces, metrics, logs

> .OpenTelemetry Collector custom resource with a client and server basicauth extension

Reviewer: ```suggestion
.OpenTelemetry Collector custom resource with client and server authentication configured for the BasicAuth extension
```

[source,yaml]
----
  config: |
    extensions:
      basicauth/server:
        htpasswd:
          file: .htpasswd <1>
        inline: |
          ${env:BASIC_AUTH_USERNAME}:${env:BASIC_AUTH_PASSWORD} <2>

      basicauth/client:
        client_auth:
          username: username <3>
          password: password <4>

    receivers:
      otlp:
        protocols:
          http:
            auth:
              authenticator: basicauth/server <5>
    exporters:
      otlp:
        auth:
          authenticator: basicauth/client <6>

service:
  extensions: [basicauth/server, basicauth/client]
  pipelines:
    traces:
      receivers: [otlp]
      exporters: [otlp]
----

> <1> The basicauth extension getting configured as a server authenticator reading credentials from a `.htpasswd` file.

Reviewer: ```suggestion
<1> Here the BasicAuth extension is configured as a server authenticator that reads credentials from an `.htpasswd` file.
```

> <2> The basicauth extension getting configured as a server authenticator reading the credentials from an inline string. This string consists of the read-in environment variables `BASIC_AUTH_USERNAME` and `BASIC_AUTH_PASSWORD`.

Reviewer: What is a `read-in environment variable`. I just want to make sure that `read-in` is a conventional term that's used by the industry. Or could you rephrase this for me so that I can understand this better?

Reviewer: ```suggestion
<2> Here the BasicAuth extension is configured as a server authenticator to read the credentials from an inline string that consists of environment variables.
```

Author: this inline field you can see in the configuration consists out of a username and password. e.g. `tom:123` username tom, password 123. In the given example we used `{$env:BASIC_AUTH_USERNAME}` as username. This will read the username from an environment variable named `BASIC_AUTH_USERNAME`.
Maybe the word `defined` is a bit better?

Reviewer: Thank you for clarifying this. I've updated my suggestion above.

Author: ```suggestion
<2> Here the BasicAuth extension is configured as a server authenticator to read the credentials from an inline string consisting of environment variables.
```

Author: Does this work too? Just to make clear that the string is build out of the env variables.

Reviewer: I updated my suggestion in https://github.com/max-cx/openshift-docs/pull/13#discussion_r1334085609 to state "inline string that consists of". Is this acceptable to you?

Author: perfect :)

> <3> The username configuration of basicauth extension getting configured as a client authenticator.

Reviewer: Not sure if this suggestion is correct. Let me know.
```suggestion
<3> The client username is configured as a client authenticator for the BasicAuth extension.
```

> <4> The password configuration of basicauth extension getting configured as a client authenticator.

Reviewer: Not sure if this suggestion is correct. Let me know.
```suggestion
<4> The client password is configured as a client authenticator for the BasicAuth extension.
```

> <5> The authenticator configuration getting assigned to an otlp receiver.

Reviewer: ```suggestion
<5> Here the authenticator configuration is assigned to an OTLP receiver.
```

> <1> Here the BasicAuth extension is configured as a server authenticator that reads credentials from an `.htpasswd` file.
> <2> Here the BasicAuth extension is configured as a server authenticator to read the credentials from an inline string that consists of environment variables.
> <3> The client username is configured as a client authenticator for the BasicAuth extension.
> <4> The client password is configured as a client authenticator for the BasicAuth extension.
> <5> Here the authenticator configuration is assigned to an OTLP receiver.
> <6> Here the authenticator configuration is assigned to an OTLP exporter.

Reviewer: I've had a second look at my own applied suggestion:
```suggestion
<1> The BasicAuth extension can be configured as a server authenticator that reads credentials from a `.htpasswd` file.
<2> The BasicAuth extension can be configured as a client authenticator to read the credentials from an inline string that consists of environment variables.
<3> The client username is configured as a client authenticator for the BasicAuth extension.
<4> The client password is configured as a client authenticator for the BasicAuth extension.
<5> The authenticator configuration can be assigned to an OTLP receiver.
<6> The authenticator configuration can be assigned to an OTLP exporter.
```

Author: I changed <2> to client.

> <6> The authenticator configuration getting assigned to an otlp exporter.

Reviewer: ```suggestion
<6> Here the authenticator configuration is assigned to an OTLP exporter.
```
